src/features/view-nft/Attributes/Attributes.tsx

```diff
@@ -0,0 +1,173 @@
+import { FC, useState, useMemo, useRef } from 'react';
+
+import { getDomainId } from '../../../lib/util';
+import { useDomainMetadata, useResize } from '../../../lib/hooks';
+import { Attribute } from '../../../lib/types/metadata';
+
+import classNames from 'classnames/bind';
+import styles from './Attributes.module.scss';
+
+const cx = classNames.bind(styles);
+
+export const NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT = {
+	MOBILE: 5,
+	TABLET: 7,
+	DESKTOP: 11,
+};
+
+interface AttributesProps {
+	zna: string;
+}
+
+export const Attributes: FC<AttributesProps> = ({ zna }) => {
+	const containerRef = useRef(null);
+	const domainId = getDomainId(zna);
+
+	const { data: metadata } = useDomainMetadata(domainId);
+
+	const [isToggled, setIsToggled] = useState<boolean>(true);
+	const [containerWidth, setContainerWidth] = useState<number>(0);
+
+	const initialVisibleAttributesCount: number = useMemo(() => {
+		const isTablet = containerWidth > 592 && containerWidth < 800;
+		const isMobile = containerWidth <= 592;
+
+		if (isMobile) return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.MOBILE;
+		if (isTablet) return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.TABLET;
+		return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.DESKTOP;
```

Comment:
You could probably handle this logic in `CSS` using `nth-child` selectors, e.g.
```
if section is not expanded
    if we're in the tablet breakpoint
        hide all children after n
    if we're in the tablet breakpoint
        hide all children after n
```
That way this component will just need to keep track of `is-expanded` and CSS does the rest. May be missing something here so take that advice with a grain of salt!